sql/core/src/main/scala/org/apache/spark/sql/execution/command/CreateSQLFunctionCommand.scala

```diff
@@ -367,6 +373,61 @@ case class CreateSQLFunctionCommand(
     }
   }
 
+    /**
+   * Check if the given plan contains cyclic function references.
+   */
+  private def checkCyclicFunctionReference(
+      catalog: SessionCatalog,
+      identifier: FunctionIdentifier,
+      plan: LogicalPlan): Unit = {
+
+    def checkPlan(plan: LogicalPlan, path: Seq[FunctionIdentifier]): Unit = {
+      plan.foreach {
+        case u @ UnresolvedTableValuedFunction(nameParts, arguments, _) =>
+          val funcId = nameParts.asFunctionIdentifier
+          val info = catalog.lookupFunctionInfo(funcId)
+          if (isSQLFunction(info.getClassName)) {
+            val f = withPosition(u) {
+              catalog.lookupTableFunction(funcId, arguments).asInstanceOf[SQLTableFunction]
+            }
+            // Check cyclic reference using qualified function names.
+            val newPath = path :+ f.function.name
+            if (f.function.name == name) {
```

Comment:
shall we consider case sensitivity?

Reply:
and is this `name` qualified?

Reply:
Function names are always case insensitive, and the name here is qualified. I've added a few more tests to cover both scenarios.